streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecorderTest.java

```diff
@@ -279,449 +310,335 @@ public void shouldThrowIfDbToAddWasAlreadyAddedForOtherSegment() {
-            blockCacheDataHitRatioSensor,
-            blockCacheIndexHitRatioSensor,
-            blockCacheFilterHitRatioSensor,
-            bytesWrittenDuringCompactionSensor,
-            bytesReadDuringCompactionSensor,
-            compactionTimeAvgSensor,
-            compactionTimeMinSensor,
-            compactionTimeMaxSensor,
-            numberOfOpenFilesSensor,
-            numberOfFileErrorsSensor
-        );
+        verifyNoInteractions(bytesWrittenToDatabaseSensor,
+                bytesReadFromDatabaseSensor,
+                memtableBytesFlushedSensor,
+                memtableHitRatioSensor,
+                memtableAvgFlushTimeSensor,
+                memtableMinFlushTimeSensor,
+                memtableMaxFlushTimeSensor,
+                writeStallDurationSensor,
+                blockCacheDataHitRatioSensor,
+                blockCacheIndexHitRatioSensor,
+                blockCacheFilterHitRatioSensor,
+                bytesWrittenDuringCompactionSensor,
+                bytesReadDuringCompactionSensor,
+                compactionTimeAvgSensor,
+                compactionTimeMinSensor,
+                compactionTimeMaxSensor,
+                numberOfOpenFilesSensor,
+                numberOfFileErrorsSensor);
     }
 
     @Test
     public void shouldCorrectlyHandleHitRatioRecordingsWithZeroHitsAndMisses() {
-        reset(statisticsToAdd1);
         recorder.addValueProviders(SEGMENT_STORE_NAME_1, dbToAdd1, cacheToAdd1, statisticsToAdd1);
-        expect(statisticsToAdd1.getHistogramData(anyObject())).andStubReturn(new HistogramData(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0L, 0L, 0.0));
-        expect(statisticsToAdd1.getAndResetTickerCount(anyObject())).andStubReturn(0L);
-        replay(statisticsToAdd1);
-        memtableHitRatioSensor.record(0, 0L);
-        blockCacheDataHitRatioSensor.record(0, 0L);
-        blockCacheIndexHitRatioSensor.record(0, 0L);
-        blockCacheFilterHitRatioSensor.record(0, 0L);
-        replay(memtableHitRatioSensor);
-        replay(blockCacheDataHitRatioSensor);
-        replay(blockCacheIndexHitRatioSensor);
-        replay(blockCacheFilterHitRatioSensor);
+        when(statisticsToAdd1.getHistogramData(any())).thenReturn(new HistogramData(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0L, 0L, 0.0));
+        when(statisticsToAdd1.getAndResetTickerCount(any())).thenReturn(0L);
 
         recorder.record(0L);
 
-        verify(memtableHitRatioSensor);
-        verify(blockCacheDataHitRatioSensor);
-        verify(blockCacheIndexHitRatioSensor);
-        verify(blockCacheFilterHitRatioSensor);
+        verify(memtableHitRatioSensor).record(0d, 0L);
+        verify(blockCacheDataHitRatioSensor).record(0d, 0L);
+        verify(blockCacheIndexHitRatioSensor).record(0d, 0L);
+        verify(blockCacheFilterHitRatioSensor).record(0d, 0L);
+        verify(statisticsToAdd1, times(2)).getHistogramData(any());
+        verify(statisticsToAdd1, times(17)).getAndResetTickerCount(any());
```

Comment:
Those verifications are not needed since you have already done them in `shouldRecordStatisticsBasedMetrics()`. This method tests the computation of the hit ratio and not the retrieval of the single values.

Reply:
done.